lock/src/main/java/com/auth0/android/lock/PasswordlessLockActivity.java

```diff
@@ -0,0 +1,236 @@
+        initLockUI();
+    }
+
+    private boolean isLaunchConfigValid() {
+        options = getIntent().getParcelableExtra(Lock.OPTIONS_EXTRA);
+        if (options == null) {
+            Log.e(TAG, "You need to specify the com.auth0.android.lock.Options in the Lock.OPTIONS_EXTRA of the Intent for PasswordlessLockActivity to launch. " +
+                    "Use com.auth0.android.lock.PasswordlessLock.Builder to generate one.");
+            throw new IllegalArgumentException("Missing com.auth0.android.lock.Options in intent");
+        }
+
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (passwordlessForm != null && passwordlessForm.onBackPressed()) {
+            return;
+        }
+        if (options != null && options.isClosable()) {
+            Intent intent = new Intent(Lock.CANCELED_ACTION);
+            LocalBroadcastManager.getInstance(PasswordlessLockActivity.this).sendBroadcast(intent);
+            return;
+        }
+        super.onBackPressed();
+    }
+
+    /**
+     * Show the LockUI with all the panels and custom widgets.
+     */
+    private void initLockUI() {
+        passwordlessForm = new PasswordlessFormView(PasswordlessLockActivity.this, lockBus, options.passwordlessMode());
+        rootView.addView(passwordlessForm, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+    }
+
+    private void deliverResult(Authentication result) {
+        Intent intent = new Intent(Lock.AUTHENTICATION_ACTION);
+        intent.putExtra(Lock.ID_TOKEN_EXTRA, result.getToken().getIdToken());
+        intent.putExtra(Lock.ACCESS_TOKEN_EXTRA, result.getToken().getAccessToken());
+        intent.putExtra(Lock.REFRESH_TOKEN_EXTRA, result.getToken().getRefreshToken());
+        intent.putExtra(Lock.TOKEN_TYPE_EXTRA, result.getToken().getType());
+        intent.putExtra(Lock.PROFILE_EXTRA, result.getProfile());
+
+        //TODO: Check if sendBroadcast works on background
+        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
+
+        finish();
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Log.d(TAG, "OnActivityResult called with intent: " + data);
+        processIncomingIntent(data);
+
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        Log.d(TAG, "OnNewIntent called with intent: " + intent);
+        processIncomingIntent(intent);
+
+        super.onNewIntent(intent);
+    }
+
+    private void processIncomingIntent(Intent intent) {
+        if (options != null && options.passwordlessMode() == PasswordlessMode.EMAIL_LINK) {
+            String code = intent.getData().getQueryParameter("code");
+            if (code == null || code.isEmpty() || lastPasswordlessEmailOrNumber == null) {
+                Log.w(TAG, "Code was invalid.");
```

Comment:
So if we open a link that is invalid we'll leave the `PasswordlessLockActivity` alive and the user has to go back manually?.

Also what will the user see in this case?